source/dotnet/wholesale-api/CalculationResults/CalculationResults.IntegrationTests/Infrastructure/RequestCalculationResult/RequestCalculationResultRetrieverTests.cs

```diff
@@ -0,0 +1,715 @@
+﻿// Copyright 2020 Energinet DataHub A/S
+//
+// Licensed under the Apache License, Version 2.0 (the "License2");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Globalization;
+using Energinet.DataHub.Core.Databricks.SqlStatementExecution.Internal;
+using Energinet.DataHub.Core.TestCommon.AutoFixture.Attributes;
+using Energinet.DataHub.Wholesale.CalculationResults.Infrastructure.RequestCalculationResult;
+using Energinet.DataHub.Wholesale.CalculationResults.Infrastructure.SqlStatements.DeltaTableConstants;
+using Energinet.DataHub.Wholesale.CalculationResults.IntegrationTests.Fixtures;
+using Energinet.DataHub.Wholesale.CalculationResults.Interfaces.CalculationResults;
+using Energinet.DataHub.Wholesale.CalculationResults.Interfaces.CalculationResults.Model.EnergyResults;
+using Energinet.DataHub.Wholesale.Common.Databricks.Options;
+using Energinet.DataHub.Wholesale.Common.Models;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Moq;
+using NodaTime;
+using Xunit;
+
+namespace Energinet.DataHub.Wholesale.CalculationResults.IntegrationTests.Infrastructure.RequestCalculationResult;
+
+public class RequestCalculationResultRetrieverTests : IClassFixture<DatabricksSqlStatementApiFixture>
+{
+    private const string BatchId = "019703e7-98ee-45c1-b343-0cbf185a47d9";
+    private const string FirstQuantity = "1.111";
+    private const string FirstQuantityFirstCorrection = "1.222";
+    private const string FirstQuantitySecondCorrection = "1.333";
+    private const string FirstQuantityThirdCorrection = "1.444";
+    private const string SecondQuantity = "2.222";
+    private const string SecondQuantityFirstCorrection = "2.333";
+    private const string SecondQuantitySecondCorrection = "2.444";
+    private const string SecondQuantityThirdCorrection = "2.555";
+    private const string ThirdQuantity = "3.333";
+    private const string FourthQuantity = "4.444";
+    private const string FourthQuantityThirdCorrection = "4.555";
+    private const string GridAreaCode = "301";
+    private readonly DatabricksSqlStatementApiFixture _fixture;
+
+    public RequestCalculationResultRetrieverTests(DatabricksSqlStatementApiFixture fixture)
+    {
+        _fixture = fixture;
+    }
+
+    [Theory]
+    [InlineAutoMoqData]
+    public async Task GetRequestCalculationResultAsync_RequestFromGridOperatorTotalProduction_ReturnsCorrectResult(
```

Comment:
```suggestion
    public async Task GetRequestCalculationResultAsync_WhenRequestFromGridOperatorTotalProduction_ReturnsCorrectResult(
```

Added when to _xx_. This is the naming convention wholesale has